src/passes/SimplifyGlobals.cpp

```diff
@@ -673,11 +673,12 @@ struct SimplifyGlobals : public Pass {
     // go, as well as applying them where possible.
     for (auto& global : module->globals) {
       if (!global->imported()) {
+        // Apply globals to this value, which may turn it into a constant we can
+        // further propagate, or it may already have been one.
+        applyGlobals(global->init);
```

Comment:
This looks like a bug fix. Which test case does it affect?

Reply:
It's not a bugfix, see the second paragraph above: This makes us slightly more efficient in SimplifyGlobals, but we always did it.

Reply:
Testcase: https://github.com/WebAssembly/binaryen/pull/6287/files#diff-b20b12708a3cf7de01646f83ed600c7dc546c88505309a98ecb19e1b3f9eda65R35